src/fhirtypes/common.ts

```diff
@@ -102,7 +109,10 @@ export function replaceReferences(rule: FixedValueRule, tank: FSHTank, resolve:
       // If the instance has a rule setting id, that overrides instance.id
       const idRule = instance.rules.find(r => r.path === 'id');
       const id = idRule?.fixedValue ?? instance.id;
-      rule.fixedValue.reference = `${instanceSD.type}/${id}`;
+      clone = new FixedValueRule(rule.path);
+      clone.fixedValue = cloneDeep(rule.fixedValue);
```

Comment:
I did it this way because TypeScript yelled at me for doing it the other way, since if you `cloneDeep(rule)` it can't figure out that `rule.fixedValue` is a `FshReference`. But I forgot about the `as` syntax in TS so I can update it to use that.